packages/eslint-plugin/src/rules/no-unnecessary-type-parameters.ts

```diff
@@ -323,8 +323,11 @@ function collectTypeParameterUsageCounts(
   }
 
   function incrementTypeUsages(type: ts.Type): number {
-    const count = (typeUsages.get(type) ?? 0) + 1;
-    typeUsages.set(type, count);
+    const typeWithoutArguments = checker
+      .typeToString(type)
+      .replaceAll(/<.+?>/g, '');
```

Comment:
Just commenting while it's in draft: we really, really try to avoid `typeToString`. Types can look exactly the same but be in two different places, and/or might only differ by whitespace.

```ts
export function one() {
  interface Box {
    value: ` ${" " | ""}`;
  }

  const box: Box = { value: " " };

  return box.value;
}

export function two() {
  interface Box {
    value: ` ${" " | ""}`;
  }

  const box: Box = { value: " " };

  return box.value;
}
```

https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBAMwK4DsDGMCWEVx8ACgEo4BvAKDjkxRmCgQEM1g4AhCEMyquAN0YAbJMABccAAZwAJKQBEcBQB9FcgL4SA3DzXkeaHAGd4AI07iOXALxl+QkeIUK12nlGAwkUXGZAA6AWFgbV1yUEhYRFQMbFwYAHcIYm4qGjoGZlZLFN5Ah0kZeUU4FTl1LR09KgMUYzhfC044G1I7IMdilyq4d09ves4A+2DyNSA